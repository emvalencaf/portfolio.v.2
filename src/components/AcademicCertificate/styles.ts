import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;