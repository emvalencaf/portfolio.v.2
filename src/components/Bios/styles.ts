import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
