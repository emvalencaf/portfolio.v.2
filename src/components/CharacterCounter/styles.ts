import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${() => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-items: end;
	`}
`;
