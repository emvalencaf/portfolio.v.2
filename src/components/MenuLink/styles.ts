import styled, { css } from 'styled-components';

export const Link = styled.a`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;

		color: ${theme.colors.primaryColor};
		text-decoration: none;
		transition: all 350ms ease-in;

		&:hover {
			color: ${theme.colors.secondaryColor};
			transform: scale(1.1);
		}
	`}
`;
