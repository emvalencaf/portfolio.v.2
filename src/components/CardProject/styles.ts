import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const Wrapper = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: space-between;
		height: ${theme.frameSizes.medium};
		width: ${theme.frameSizes.smallMedium};
		border-radius: 2.9rem;
		padding: ${theme.spacings.small};
		background-color: ${theme.colors.gray1};
		transition: ${theme.transitions.normal};
		opacity: 0.4;
		&:focus,
		&:hover {
			opacity: 1;
			box-shadow: 0 0 ${theme.spacings.medium} ${theme.spacings.tiny} ${theme.colors.gray6};
			filter: brightness(110%);
		}
	`}
`;

export const PictureContainer = styled.picture`
	${({ theme }) => css`
		@keyframes photo {
			0% {
				transform: scale(0);
			}
			50% {
				transform: scale(1.1);
			}
			100% {
				transform: scale(1);
			}
		}

		display: flex;
		align-items: center;
		justify-content: center;
		width: 100%;
		transform: scale(0);
		transition: all ${theme.transitions.normal} ease-in-out;
		animation: photo ${theme.transitions.slowest};
		animation-fill-mode: forwards;

		> img {
			width: 100%;
			height: 100%;
			margin: auto;
			border-radius: 2rem;
		}
	`}
`;

export const ContentContainer = styled.p`
	${({ theme }) => css`
		@keyframes content {
			0% {
				transform: scale(0);
			}
			50% {
				transform: scale(1.1);
			}
			100% {
				transform: scale(1);
			}
		}
		transform: scale(0);
		transition: all ${theme.transitions.normal} ease-in-out;
		animation: content ${theme.transitions.slowest};
		animation-fill-mode: forwards;
	`}
`;

export const HeaderCardContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: space-between;
		width: 100%;
		background-color: ${theme.colors.gray4};
		border-radius: 2rem;
		opacity: 0.7;
		z-index: ${theme.layers.layer3};

		& ${HeadingContainer} {
			text-align: center;
			font-size: ${theme.fonts.sizes.normal};
			color: ${theme.colors.gray6};
			margin: 0px;
			padding: ${theme.spacings.xxsmall};
		}

		& svg {
			height: ${theme.fonts.sizes.medium};
			width: ${theme.fonts.sizes.medium};
			transition: all ${theme.transitions.normal} ease-in-out;
			margin: ${theme.spacings.small};
			&:hover {
				color: ${theme.colors.primary};
			}
		}

		> span > svg {
			@keyframes rotate {
				0% {
					transform: rotate(90deg);
				}
				100% {
					transform: rotate(180deg);
				}
				100% {
					transform: rotate(360deg);
				}
			}
			cursor: pointer;
			&:hover {
				animation: rotate ${theme.transitions.slowest} infinite;
			}
		}
	`}
`;

export const ButtonsContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		gap: ${theme.spacings.small};
		> a {
			background-color: ${theme.colors.gray6};
			color: ${theme.colors.white};
			font-size: ${theme.fonts.sizes.normal};
			padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
			cursor: pointer;
			border-radius: ${theme.spacings.tiny};
			transition: all ${theme.transitions.fast} ease-in-out;
			display: flex;
			align-items: center;
			justify-content: center;
			text-decoration: none;

			> svg {
				width: 2rem;
				height: 2rem;
				margin-left: 1rem;
				margin-right:1rem;
			}

			&:focus {
				outline: none;
				box-shadow: 0 0 ${theme.spacings.tiny} ${theme.colors.primary};
				filter: brightness(110%);
			}

			&:hover {
				filter: brightness(90%);
			}
		}
	`}
`;