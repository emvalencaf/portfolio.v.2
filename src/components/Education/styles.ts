import styled, { css } from 'styled-components';
import { Button } from '../Button/styles';

export const Wrapper = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		gap: ${theme.spacings.mediumSmall};
	`}
`;

export const OrderList = styled.ol`
	${({ theme }) => css`
		margin: 0;
		padding: 0;
		list-style: none;
		max-width: 400px;

		> li {
			padding: 0px ${theme.spacings.mediumSmall};
		}

		> li:has(> button:focus) {
			list-style: disc;
			color: ${theme.colors.primary};
		}

		& ${Button} {
			align-items: flex-start;
			justify-content: flex-start;
			color: ${theme.colors.primary};
			background-color: transparent;
			font-size: ${theme.fonts.sizes.small};
			text-align: justify;
			border: none;
			outline: none;
			width: 100%;
		}

		& ${Button}:active,
		& ${Button}:focus,
		& ${Button}:hover {
			box-shadow: none;
		}

	`}
`;

export const Content = styled.div`

`;