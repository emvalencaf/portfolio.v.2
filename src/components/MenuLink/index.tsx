// styles
import Link from 'next/link';
import * as Styled from './styles';

// types
export type MenuLinkProps = {
	children: React.ReactNode;
	link: string;
	newTab?: boolean;
};

const MenuLink = ({ children, link, newTab = false }: MenuLinkProps) => {
	const target = newTab ? "_blank" : "_self";
	const hostname = location.hostname;
	const rel = link.includes(hostname) || link.match(/^[\/#]/) ? "internal" : "external";
	const nextLink = link.match(/^\//) ? true : false;

	if (nextLink) return (
		<Link href={link} legacyBehavior passHref>
			<Styled.Link target={target} rel={rel}>
				{children}
			</Styled.Link>
		</Link>
	);

	return (
		<Styled.Link href={link} target={target} rel={rel}>
			{children}
		</Styled.Link>
	)
};

export default MenuLink;
