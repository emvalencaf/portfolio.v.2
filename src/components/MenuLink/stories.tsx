import { Meta, Story } from "@storybook/react/types-6-0";
import MenuLink, { MenuLinkProps } from ".";

export default {
	title: "MenuLink",
	component: MenuLink,
	args: {
		link: "/",
		text: "click here",
		newTab: false,
	},
	argTypes: {
		link: { type: "string" },
		text: { type: "string" },
		newTab: { type: "boolean" },
	},
} as Meta<MenuLinkProps>;

export const Template: Story<MenuLinkProps> = (args) => {
	return (
		<div>
			<MenuLink {...args} />
		</div>
	);
};
