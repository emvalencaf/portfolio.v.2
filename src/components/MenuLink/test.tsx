import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

// react component to be test
import MenuLink, { MenuLinkProps } from '.';

const mock: MenuLinkProps = {
	children: "click here",
	link: "/",
};

describe('<MenuLink />', () => {
	it("should render MenuLink component with a anchor element", () => {
		renderTheme(<MenuLink link="#home">Home</MenuLink>);
		expect(screen.getByRole("link", { name: "Home" })).toHaveAttribute("href", "#home");
	});

	it("should render a MenuLink component with nextlink component", () =>{
		renderTheme(<MenuLink link='/home'>Home</MenuLink>);
		expect(screen.getByRole("link", { name: "Home" })).toHaveAttribute("href", "/home");
	});

	it("should render MenuLink component with a internal link", () => {
		renderTheme(<MenuLink link='https://localhost:3000/home'>Home</MenuLink>);

		expect(screen.getByRole("link", { name: "Home" })).toHaveAttribute("rel","internal");
	});

	it("should render MenuLink component with a external link that will open a new tab", () => {
		renderTheme(<MenuLink link='https://google.com' newTab>Home</MenuLink>);

		expect(screen.getByRole("link", { name: "Home" })).toHaveAttribute("target","_blank");
		expect(screen.getByRole("link", { name: "Home" })).toHaveAttribute("rel","external");
	});

	it("should match a snapshot", () => {
		const { container } = renderTheme(<MenuLink link="#" >Home</MenuLink>);

		expect(container).toMatchSnapshot();
	})
});
