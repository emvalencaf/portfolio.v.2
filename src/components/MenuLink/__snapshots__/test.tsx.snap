// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match a snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #0A1128;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
}

.c0:hover {
  color: #dc143c;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div>
  <a
    class="c0"
    href="#"
    rel="external"
    target="_self"
  >
    Home
  </a>
</div>
`;
